components/kyma-environment-broker/internal/event-hub/event_hub_step_test.go

```diff
@@ -0,0 +1,445 @@
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/provisioning/input"
+	inputAutomock "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/provisioning/input/automock"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+const kymaVersion = "1.10"
+
+func fixLogger() logrus.FieldLogger {
+	return logrus.StandardLogger()
+}
+
+// ensure the fake client is implementing the interface
+var _ azure.EventhubsInterface = (*FakeNamespaceClient)(nil)
+
+/// A fake client for Azure EventHubs Namespace handling
+type FakeNamespaceClient struct {
+	persistEventhubsNamespaceError error
+	listError                      error
+	resourceGroupError             error
+}
+
+func (nc *FakeNamespaceClient) GetEventhubAccessKeys(ctx context.Context, resourceGroupName string, namespaceName string, authorizationRuleName string) (result eventhub.AccessKeys, err error) {
+	return eventhub.AccessKeys{
+		PrimaryConnectionString: ptr.String("Endpoint=sb://name/;"),
+	}, nc.listError
+}
+
+func (nc *FakeNamespaceClient) CreateResourceGroup(ctx context.Context, config *azure.Config, name string) (resources.Group, error) {
+	return resources.Group{
+		Name: ptr.String("my-resourcegroup"),
+	}, nc.resourceGroupError
+}
+
+func (nc *FakeNamespaceClient) CreateNamespace(ctx context.Context, azureCfg *azure.Config, groupName, namespace string) (*eventhub.EHNamespace, error) {
+	return &eventhub.EHNamespace{
+		Name: ptr.String(namespace),
+	}, nc.persistEventhubsNamespaceError
+}
+
+func NewFakeNamespaceClientCreationError() azure.EventhubsInterface {
+	return &FakeNamespaceClient{persistEventhubsNamespaceError: fmt.Errorf("error while creating namespace")}
+}
+
+func NewFakeNamespaceClientListError() azure.EventhubsInterface {
+	return &FakeNamespaceClient{listError: fmt.Errorf("cannot list namespaces")}
+}
+
+func NewFakeNamespaceResourceGroupError() azure.EventhubsInterface {
+	return &FakeNamespaceClient{resourceGroupError: fmt.Errorf("cannot create resource group")}
+}
+
+func NewFakeNamespaceClientHappyPath() azure.EventhubsInterface {
+	return &FakeNamespaceClient{}
+}
+
+// ensure the fake client is implementing the interface
+var _ azure.HyperscalerProvider = (*fakeHyperscalerProvider)(nil)
+
+type fakeHyperscalerProvider struct {
+	client azure.EventhubsInterface
+	err    error
+}
+
+func (ac *fakeHyperscalerProvider) GetClient(config *azure.Config) (azure.EventhubsInterface, error) {
+	return ac.client, ac.err
+}
+
+func NewFakeHyperscalerProvider(client azure.EventhubsInterface) azure.HyperscalerProvider {
+	return &fakeHyperscalerProvider{
+		client: client,
+		err:    nil,
+	}
+}
+
+func NewFakeHyperscalerProviderError() azure.HyperscalerProvider {
+	return &fakeHyperscalerProvider{
+		client: nil,
+		err:    fmt.Errorf("ups ... "),
```

Comment:
I assume you do not expect this to happen

Reply:
I expect this to happen, because it triggers the edge case in the eventhub step.

It is used in the test [here](https://github.com/anishj0shi/compass/blob/00ea98a7139d8d0c954ee8899f43dd2ab1a22bb5/components/kyma-environment-broker/internal/event-hub/event_hub_step_test.go#L222)

It is testing the following lines:

```go
	namespaceClient, err := p.hyperscalerProvider.GetClient(azureCfg)
	if err != nil { <- it produces this error
		// internal error, repeating doesn't solve the problem
		errorMessage := fmt.Sprintf("Failed to create Azure EventHubs client: %v", err)
		return p.operationManager.OperationFailed(operation, errorMessage)
	}
```

Reply:
well then lets use some proper message here :D

Reply:
all ups are more detailed now :-D